OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -51,24 +52,38 @@ public Fly(Actor self, Target t, WPos? initialTargetPosition = null, Color? targ
 			this.minRange = minRange;
 		}
 
-		public static void FlyToward(Actor self, Aircraft aircraft, int desiredFacing, WDist desiredAltitude, int turnSpeedOverride = -1)
+		public static bool FlyToward(Actor self, Aircraft aircraft, int desiredFacing, WDist desiredAltitude, int turnSpeedOverride = -1, bool isVTOL = false)
+		{
+			var move = aircraft.Info.CanHover ? aircraft.FlyStep(desiredFacing) : aircraft.FlyStep(aircraft.Facing);
+			return FlyToward(self, aircraft, move, desiredFacing, desiredAltitude, turnSpeedOverride, isVTOL);
+		}
+
+		// Note: isVTOL is meant to only refer to actual take-off or land, not mere terrain-height-induced altitude changes.
+		// This is to avoid VTOLs constantly 'braking' on terrain height changes.
+		public static bool FlyToward(Actor self, Aircraft aircraft, WVec move, int desiredFacing, WDist desiredAltitude, int turnSpeedOverride = -1, bool isVTOL = false)
```

Comment:
Can we please replace `bool isVTOL` with `MovementType types`?

This turns a hack into a feature and creates a relatively coherant external interface.
We can move the facing tick logic inside this method for `MovementType.Turn`.